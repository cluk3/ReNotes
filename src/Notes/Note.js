import React from "react";

const humanFriendlyDate = date => "Monday";
const createExcerptFromText = text => {
  const body = text.split("\n")[1];
  return body ? body.slice(0, 80) : "No additional text";
};
const createTitleFromText = text => text.split("\n")[0] || "New Note";

const Note = ({ text, creationDate, handleNoteClick, selected }) => {
  const title = createTitleFromText(text);
  const excerpt = createExcerptFromText(text);
  const creationTime = humanFriendlyDate(creationDate);
  return (
    <div
      onClick={handleNoteClick}
      style={{
        backgroundColor: selected ? "grey" : "white"
      }}
    >
      <h2>{title}</h2>
      <span>{creationTime}</span>
      <p>{excerpt}</p>
    </div>
  );
};

export default Note;
