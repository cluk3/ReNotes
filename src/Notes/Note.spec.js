import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Note from "./Note";
describe("Note", () => {
  it("should render correctly", () => {
    const output = shallow(
      <Note
        text={`title\nexcerpt`}
        lastModified={2314324124123}
        handleNoteClick={jest.fn()}
        selected={false}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it("should show the correct title", () => {
    const output = shallow(
      <Note
        text="test"
        lastModified={2314324124123}
        handleNoteClick={jest.fn()}
        selected={false}
      />
    );
    expect(output.find("h2").text()).toEqual("test");
  });

  it("should show the correct creation date info", () => {
    const output = shallow(
      <Note
        text="test"
        lastModified={2314324124123}
        handleNoteClick={jest.fn()}
        selected={false}
      />
    );
    expect(output.find("span").text()).toEqual("Monday");
  });

  it("should show the default excerpt when note has no body", () => {
    const output = shallow(
      <Note
        text="test"
        lastModified={2314324124123}
        handleNoteClick={jest.fn()}
        selected={false}
      />
    );
    expect(output.find("p").text()).toEqual("No additional text");
  });

  it("should show the correct title and excerpt with new line", () => {
    const output = shallow(
      <Note
        text={`test title\ntest excerpt`}
        lastModified={2314324124123}
        handleNoteClick={jest.fn()}
        selected={false}
      />
    );
    expect(output.find("h2").text()).toEqual("test title");
    expect(output.find("p").text()).toEqual("test excerpt");
  });

  it("should have background grey when selected", () => {
    const output = shallow(
      <Note
        text={`title\nexcerpt`}
        lastModified={2314324124123}
        handleNoteClick={jest.fn()}
        selected={true}
      />
    );
    expect(output.first().prop("style").backgroundColor).toEqual("grey");
  });
});
