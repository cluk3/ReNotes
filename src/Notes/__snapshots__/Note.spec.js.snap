// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note works 1`] = `
.c1 {
  font-size: 14px;
  padding: 1em 1em 1em 0;
  border-bottom: #dedede solid 1px;
}

.c2 {
  font-size: 15px;
  margin: 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  width: 100%;
  line-height: 1.4;
}

.c4 {
  padding-right: 1em;
  white-space: nowrap;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

.c5 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 {
  background-color: #fbdb7c;
  padding-left: 1em;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
    >
      Title
    </h2>
    <div
      class="c3"
    >
      <span
        class="c4"
      >
        Mocked
      </span>
      <span
        class="c5"
      >
        excerpt
      </span>
    </div>
  </div>
</div>
`;
